### DISCUSSION TITLE
Why does the UI designer need to scan the output directory? Cause it to keep looping "loading designer you can continue working"

### DISCUSSION THREAD

**Discussion by RuoYear** [dotnet/winforms#6460] [Q&A] [answered]
* .NET Core Version: .NET 6.0.1

* Have you experienced this same bug with .NET Framework?: NO

Hello, I am using visual studio 2022, and I upgraded my previous .NET Framework project to .NET 6, but when I use the designer, it just keeps prompting "The designer is loading, you can use the designer in the background Continue to work while loading".
But when I rebuild a similar project step by step, it can work normally for a while, but most of the time it still stays at "Loading", and only "Creating VSCodeDomDesignerLoader" is displayed in the output log.

When I found the location where the log can be output in the options, I discovered why it scans the output file directory?
I placed the output directory in a separate partition due to other requirements. As a result, it scanned "System Volume Information" and was denied access. Why did the designer do this when opening it?
When I found this question and changed the output directory to a subfolder again, the designer worked. How to adjust the designer so that it will not affect the operation of the target folder in the root directory of the partition?

**Comment by Shyam-Gupta** [comment, marked as answer]
@RuoYear Designer loads project output dll and its references so as to support user's custom controls.
I will look into the scenario when designer process doesn't have access to project output assemblies and get back. Thanks.

**Reply by RuoYear** [reply]
Thanks for your answer, I also found that it seems to be for loading custom controls;
Reading all files and folders in the directory in this way also brings some special problems. For example, the program contains webview2, and it will generate a folder (*.exe.WebView2) when it runs. At this time, I want to open the designer Modify it and it will also give an error like "System Volume Information".
But in Framework, this problem doesn't seem to arise either, only the necessary files are read, not to access other things that don't belong to it. (and very fast)
Until there is a better way, is it possible to add a "catch" to the loading process and only access files that only belong to it? As in the past, without this control shows a grey bottom window and recommends retrying after the build, rather than getting stuck at "loading" until an error is found in the console ;-)

**Reply by Shyam-Gupta** [reply]
In [17.1 Preview 2](https://devblogs.microsoft.com/visualstudio/visual-studio-17-1-preview-2-is-now-available/) we have made improvements in this scenario and only a small subset of files, which are essential for the designer, will get copied. It will reduce the overall time taken to launch the designer especially for large projects\solutions.

Hence I request you to try the scenario in 17.1 Preview 2 or newer versions and share your feedback. Thanks.

**Reply by RuoYear** [reply]
Thanks your DEBUG. When I update to the official version(17.1.0) and see this message, this problem has been solved. Now it can open normally, at least it won't wait endlessly as before.

However, I don't know whether it will continue to happen in other cases. After discovering this problem, I put the generated files in a separate folder and the code affecting the designer in a separate DLL. When this DLL is occupied, the designer can still be opened smoothly.

**Reply by Shyam-Gupta** [reply]
Thank you for trying the 17.1.0 version and I am glad that it helped in this scenario.
With 17.1.0 release, designer only copies selected binary files to a temporary location instead of copying all files from project output directory. Let me know if you still see issues in any scenario.

FYI: @merriemcgaw
